### PR TITLE
feat(payment): PI-2493 Add paymentHumanVerificationHandler method to …

### PR DESCRIPTION
**Pull request by bc-sebastianszafraniec**
…Payment Integration Service

## What?
Added paymentHumanVerificationHandler method to Payment Integration Service

## Testing / Proof
Tested manually

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

File: packages/core/src/payment-integration/default-payment-integration-service.ts [hunk 1]
```diff
@@ -285,4 +287,12 @@ export default class DefaultPaymentIntegrationService implements PaymentIntegrat
     async validateCheckout(checkout?: Checkout, options?: RequestOptions): Promise<void> {
         await this._checkoutValidator.validate(checkout, options);
     }
+
+    async handle(errorOrId: Error | string, key?: string): Promise<PaymentAdditionalAction> {
```
Comment (bc-nick) [inline packages/core/src/payment-integration/default-payment-integration-service.ts:291]: Need to think of a name for this function that is more specific to the action the function performs. @bc-sebastianszafraniec what do you think?

Comment (bc-sebastianszafraniec) [reply packages/core/src/payment-integration/default-payment-integration-service.ts:291]: @bc-nick yes, maybe it is better to use some more specific name. What would you say for `handlePaymentHumanVerification`

Comment (bc-nick) [reply packages/core/src/payment-integration/default-payment-integration-service.ts:291]: Much better for me than it is now

File: packages/core/src/payment-integration/default-payment-integration-service.ts [hunk 2]
```diff
@@ -285,4 +287,15 @@ export default class DefaultPaymentIntegrationService implements PaymentIntegrat
     async validateCheckout(checkout?: Checkout, options?: RequestOptions): Promise<void> {
         await this._checkoutValidator.validate(checkout, options);
     }
+
+    async handlePaymentHumanVerification(
+        errorOrId: Error | string,
+        key?: string,
+    ): Promise<PaymentAdditionalAction> {
+        if (typeof errorOrId === 'string') {
```
Comment (bc-nick) [inline packages/core/src/payment-integration/default-payment-integration-service.ts:295]: I checked that we already have the same check inside of `_paymentHumanVerificationHandler.handle` method. Why did you duplicate it here?

Comment (bc-sebastianszafraniec) [reply packages/core/src/payment-integration/default-payment-integration-service.ts:295]: Without this check I was getting Typescript error here

Comment (bc-nick) [reply packages/core/src/payment-integration/default-payment-integration-service.ts:295]: Indeed, i did catch that we have a few declarations of `handle` function. But why we separated the declaration from implementation, i do not know. Anyway thanks for your updates!

**Review by bc-nick — APPROVED** [review]

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**